components/resources/library/src/skikoMain/kotlin/org/jetbrains/compose/resources/FontResources.skiko.kt

```diff
@@ -45,4 +49,26 @@ actual fun Font(resource: FontResource, weight: FontWeight, style: FontStyle): F
         }
     }
     return fontFile
+}
+
+@Composable
+actual fun Font(
+    resource: FontResource,
+    weight: FontWeight,
+    style: FontStyle,
+    variationSettings: FontVariation.Settings,
+): Font {
+    val resourceReader = LocalResourceReader.currentOrPreview
+    val fontFile by rememberResourceState(resource, weight, style, variationSettings, { defaultEmptyFont }) { env ->
+        val path = resource.getResourceItemByEnvironment(env).path
+        val settingsKey = variationSettings.settings
+            .sortedBy { it::class.simpleName }
+            .joinToString(",") { it::class.simpleName + "_" + it.hashCode() }
```

Comment:
-  `it.hashCode()` can cause collisions
- It seems that Setting.axisName is [unique](https://github.com/JetBrains/compose-multiplatform-core/blob/2cbd13c3c91e796df47bed6e21b7b225cf2c851e/compose/ui/ui-text/src/commonMain/kotlin/androidx/compose/ui/text/font/FontVariation.kt#L234) and can be used as the key

Reply:
settings can not contain same setting class, so I decided it is better then axisName. because different settings may contain the same value

Reply:
> settings can not contain same setting class

Hmm, I see that they can:
<img width="629" alt="image" src="https://github.com/user-attachments/assets/5792b53d-2e55-4e48-98af-f2b9842d10f0" />

If I understand it correctly, we should replace `it::class.simpleName` by `axisName`

> because different settings may contain the same value

Indeed, I didn't take it into account. But `hashCode()` is unreliable. It seems we should use something like `toVariationValue(Density(1f))`:

```
variationSettings.settings
            .sortedBy { it.axisName }
            .joinToString(",") { it.axisName  + "_" + it.toVariationValue(Density(1f)) }
```

Reply:
Another option is to replace `key: String` by `key: Any` in `fontCache.getOrLoad`, and add equality checks

Reply:
Done. with unit tests now